.github/workflows/ci.yml

```diff
@@ -0,0 +1,170 @@
+      - run: rustup component add clippy
+      - uses: Swatinem/rust-cache@v2
+      - uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all-targets
+
+  # Builds
+  build-node-binaries:
+    name: Build node binaries
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+
+      - uses: Swatinem/rust-cache@v2
+      - name: Compile
+        run: cargo build --release --bin node_reader --bin node_writer
+      - name: Upload binaries to cache
+        uses: actions/cache/save@v4
+        with:
+          path: "target/release/node_*er"
+          key: node-bin-${{ github.sha }}
+
+  build-virtual-env:
+    name: Build node bindings
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: Swatinem/rust-cache@v2
+      - uses: actions/cache@v3
+        with:
+          path: ~/.cache/pip
+          key: pip-${{ hashFiles('**/*requirements*.txt') }}
+      - name: Install venv
+        run: |
+          python -m venv /tmp/venv
+          source /tmp/venv/bin/activate
+          (pip install maturin && maturin build -m nucliadb_node_binding/Cargo.toml) &
+          /tmp/venv/bin/pip install -r test-requirements.txt -r code-requirements.txt -r nucliadb/requirements-sources.txt -r nucliadb/requirements.txt -r nucliadb/requirements-test.txt ./nucliadb
+          wait $(jobs -p)
+          pip install ./nucliadb_node_binding
+      - name: Upload virtualenv to cache
+        uses: actions/cache/save@v4
+        with:
+          path: /tmp/venv
+          key: venv-${{ github.sha }}
+  
+  # Tests
+  node-python-tests:
+    name: Node Python tests
+    runs-on: ubuntu-latest
+    needs:
+      - build-node-binaries
+      - build-virtual-env
+    steps:
+      - uses: actions/checkout@v3
+      - name: Restore venv
+        uses: actions/cache/restore@v4
+        with:
+          path: /tmp/venv
+          key: venv-${{ github.sha }}
+          fail-on-cache-miss: true
+      - name: Restore binaries
+        uses: actions/cache/restore@v4
+        with:
+          path: "target/release/node_*er"
+          key: node-bin-${{ github.sha }}
+          fail-on-cache-miss: true
+        
+      - name: Build docker images for node
+        run: mkdir builds && mv target/release/node_* builds && docker build -t europe-west4-docker.pkg.dev/nuclia-internal/nuclia/node:latest -f Dockerfile.node_prebuilt .
+        
+      - name: Run tests
+        run: /tmp/venv/bin/pytest -rfE --cov=nucliadb_node -s --tb=native -v --cov-report xml --cov-append nucliadb_node/nucliadb_node
+
+
+  node-rust-tests:
+    name: Node Rust tests
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: Swatinem/rust-cache@v2
+    - uses: actions-rs/cargo@v1
+      env:
+        RUST_BACKTRACE: 1
+      name: cargo test
+      with:
+        command: test
+        args: --workspace --all-features
+  
+  
+  nucliadb-tests:
```

Comment:
Can you add pre-checks for the python part too?

Reply:
if you want on a separate pr ofcourse

Reply:
I'll add them here so it's easier to track what has been implemented in the new pipeline, this way it's the full nucliadb_node.yml